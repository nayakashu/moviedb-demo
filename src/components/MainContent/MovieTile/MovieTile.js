import React from 'react';
import './MovieTile.scss';

export default function MovieTile() {
  return (
    <div className="movie-tile">
      <span className="popularity">
        89%
        <span className="bottom-border" />
      </span>
      <img
        src="https://image.tmdb.org/t/p/w200/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg"
        alt="Title"
      />
      <div className="movie-details">
        <h4>Fight club</h4>
        <div className="release-date">14 February 2019</div>
        <div className="description">
          A ticking-time-bomb insomniac and a slippery soap salesman channel
          primal male aggression into a shocking new form of therapy.
        </div>
      </div>
    </div>
  );
}
