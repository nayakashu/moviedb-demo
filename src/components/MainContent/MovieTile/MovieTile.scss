@import '../../../Common.scss';

.movie-tile {
  background: $header-bg-color;
  color: $text-color-light;
  position: relative;
  padding: 15px;
  box-shadow: 1px 2px 10px $shadow-color-dark;
  margin-top: 100px;

  @include border-radius(2px);

  .popularity {
    position: absolute;
    right: 15px;
    top: 25px;
    color: $highlight-green;
    font-size: 11px;
    font-weight: bold;
    padding-bottom: 3px;
    .bottom-border {
      position: absolute;
      border: 1px solid $highlight-green;
      bottom: 0px;
      width: 8px;
      left: 20%;
    }
  }

  i.ellipsis {
    position: absolute;
    right: 5px;
    top: 5px;
    font-size: 15px;
    cursor: pointer;
    color: $content-text-color;
  }

  i.btn-like {
    position: absolute;
    right: 10px;
    bottom: 10px;
    font-size: 18px;
    cursor: pointer;
    color: $heart-red;
  }

  img {
    width: 45%;
    display: inline-block;
    position: absolute;
    bottom: 25px;
    box-shadow: 1px 2px 10px $shadow-color-dark;
    @include border-radius(2px);
  }

  .movie-details {
    padding-left: 52%;
    height: 180px;

    h4 {
      font-size: 20px;
      padding-right: 25px;
    }

    .release-date {
      font-size: 11px;
      color: $content-text-color;
    }

    .description {
      margin-top: 10px;
      color: $content-text-color;
      font-size: 13px;
    }
  }
}
