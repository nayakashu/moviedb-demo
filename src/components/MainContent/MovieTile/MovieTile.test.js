import React from 'react';
import { shallow } from 'enzyme';
import MovieTile from './MovieTile';

describe('MainContent', () => {
  const movieTile = shallow(<MovieTile />);

  it('renders correctly', () => {
    expect(movieTile).toMatchSnapshot();
  });
});
