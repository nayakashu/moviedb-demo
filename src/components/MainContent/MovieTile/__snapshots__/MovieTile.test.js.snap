// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieTile />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="popularity"
        >
          89%
          <span
            className="bottom-border"
          />
        </span>,
        <img
          alt="Title"
          src="https://image.tmdb.org/t/p/w200/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg"
        />,
        <div
          className="movie-details"
        >
          <h4>
            Fight club
          </h4>
          <div
            className="release-date"
          >
            14 February 2019
          </div>
          <div
            className="description"
          >
            A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.
          </div>
        </div>,
      ],
      "className": "movie-tile",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "89%",
            <span
              className="bottom-border"
            />,
          ],
          "className": "popularity",
        },
        "ref": null,
        "rendered": Array [
          "89%",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "bottom-border",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Title",
          "src": "https://image.tmdb.org/t/p/w200/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h4>
              Fight club
            </h4>,
            <div
              className="release-date"
            >
              14 February 2019
            </div>,
            <div
              className="description"
            >
              A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.
            </div>,
          ],
          "className": "movie-details",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Fight club",
            },
            "ref": null,
            "rendered": "Fight club",
            "type": "h4",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "14 February 2019",
              "className": "release-date",
            },
            "ref": null,
            "rendered": "14 February 2019",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.",
              "className": "description",
            },
            "ref": null,
            "rendered": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.",
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="popularity"
          >
            89%
            <span
              className="bottom-border"
            />
          </span>,
          <img
            alt="Title"
            src="https://image.tmdb.org/t/p/w200/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg"
          />,
          <div
            className="movie-details"
          >
            <h4>
              Fight club
            </h4>
            <div
              className="release-date"
            >
              14 February 2019
            </div>
            <div
              className="description"
            >
              A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.
            </div>
          </div>,
        ],
        "className": "movie-tile",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "89%",
              <span
                className="bottom-border"
              />,
            ],
            "className": "popularity",
          },
          "ref": null,
          "rendered": Array [
            "89%",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "bottom-border",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Title",
            "src": "https://image.tmdb.org/t/p/w200/adw6Lq9FiC9zjYEpOqfq03ituwp.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Fight club
              </h4>,
              <div
                className="release-date"
              >
                14 February 2019
              </div>,
              <div
                className="description"
              >
                A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.
              </div>,
            ],
            "className": "movie-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Fight club",
              },
              "ref": null,
              "rendered": "Fight club",
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "14 February 2019",
                "className": "release-date",
              },
              "ref": null,
              "rendered": "14 February 2019",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.",
                "className": "description",
              },
              "ref": null,
              "rendered": "A ticking-time-bomb insomniac and a slippery soap salesman channel primal male aggression into a shocking new form of therapy.",
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
